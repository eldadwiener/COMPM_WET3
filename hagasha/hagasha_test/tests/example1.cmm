void main()
{
	write("\nMoodle Test 1 (block context)\n");
	a:int;
        b:int;

	read(a);
	read(b);
	write("should be the smaller number: ");
	if a < b then write(a);
	else write(b);
	write("\n");
	{ # Start new BLK with local var a
		a: int;
	        a = b + 256;
        	write("\na'="); write(a); write(" (should be 256 + second number) ");
    }
    write("\nat main BLK a="); write(a); write("\n");
}
