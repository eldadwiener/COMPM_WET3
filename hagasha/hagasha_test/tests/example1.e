COPYI I1 0
COPYI I2 0
COPYI I3 0
COPYI I4 0
COPYI I5 0
COPYI I6 0
COPYI I7 0
JLINK 10
HALT
PRNTC 10
PRNTC 77
PRNTC 111
PRNTC 111
PRNTC 100
PRNTC 108
PRNTC 101
PRNTC 32
PRNTC 84
PRNTC 101
PRNTC 115
PRNTC 116
PRNTC 32
PRNTC 49
PRNTC 32
PRNTC 40
PRNTC 98
PRNTC 108
PRNTC 111
PRNTC 99
PRNTC 107
PRNTC 32
PRNTC 99
PRNTC 111
PRNTC 110
PRNTC 116
PRNTC 101
PRNTC 120
PRNTC 116
PRNTC 41
PRNTC 10
ADD2I I2 I2 4
ADD2I I2 I2 4
READI I3
STORI I3 I1 0
READI I3
STORI I3 I1 4
PRNTC 115
PRNTC 104
PRNTC 111
PRNTC 117
PRNTC 108
PRNTC 100
PRNTC 32
PRNTC 98
PRNTC 101
PRNTC 32
PRNTC 116
PRNTC 104
PRNTC 101
PRNTC 32
PRNTC 115
PRNTC 109
PRNTC 97
PRNTC 108
PRNTC 108
PRNTC 101
PRNTC 114
PRNTC 32
PRNTC 110
PRNTC 117
PRNTC 109
PRNTC 98
PRNTC 101
PRNTC 114
PRNTC 58
PRNTC 32
LOADI I3 I1 0
LOADI I4 I1 4
SLETI I5 I3 I4
BNEQZ I5 82
UJUMP 85
LOADI I5 I1 0
PRNTI I5
UJUMP 87
LOADI I6 I1 4
PRNTI I6
PRNTC 10
ADD2I I2 I2 4
LOADI I7 I1 4
COPYI I8 256
ADD2I I7 I7 I8
STORI I7 I1 8
PRNTC 10
PRNTC 97
PRNTC 39
PRNTC 61
LOADI I9 I1 8
PRNTI I9
PRNTC 32
PRNTC 40
PRNTC 115
PRNTC 104
PRNTC 111
PRNTC 117
PRNTC 108
PRNTC 100
PRNTC 32
PRNTC 98
PRNTC 101
PRNTC 32
PRNTC 50
PRNTC 53
PRNTC 54
PRNTC 32
PRNTC 43
PRNTC 32
PRNTC 115
PRNTC 101
PRNTC 99
PRNTC 111
PRNTC 110
PRNTC 100
PRNTC 32
PRNTC 110
PRNTC 117
PRNTC 109
PRNTC 98
PRNTC 101
PRNTC 114
PRNTC 41
PRNTC 32
PRNTC 10
PRNTC 97
PRNTC 116
PRNTC 32
PRNTC 109
PRNTC 97
PRNTC 105
PRNTC 110
PRNTC 32
PRNTC 66
PRNTC 76
PRNTC 75
PRNTC 32
PRNTC 97
PRNTC 61
LOADI I10 I1 0
PRNTI I10
PRNTC 10
COPYI I2 I1
RETRN
HALT
