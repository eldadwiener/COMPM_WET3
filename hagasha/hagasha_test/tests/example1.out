
Moodle Test 1 (block context)
Input integer?:Input integer?:should be the smaller number: 4 

a'=262  (should be 256 + second number) 
at main BLK a=4 
Reached Halt.
